docs/design/vm_data_protection-design.md

```diff
@@ -0,0 +1,502 @@
+
+3. **Kopia Server Deployment**  
+   Create or update the Kopia server Pod and associated Kubernetes resources to ensure the server is running and accessible.
+
+4. **Credential Synchronization**  
+   Sync user credentials stored as Kubernetes Secrets to the Kopia server, enabling authenticated access for VM users.
+
+5. **Configuration Updates**  
+   Detect changes in the BSLS or BSLR resources and apply updates to the Kopia server configuration accordingly.
+
+Through these reconciliation steps, the BSLS controller ensures the Kopia server is securely configured, operational, and aligned with cluster resource definitions.
+
+### BackupStorageLocationServer CRD
+
+The **BackupStorageLocationServer (BSLS)** custom resource defines a Kopia proxy server instance in OpenShift. It acts as an intermediary between clients (e.g., VMs) and the underlying Kopia repository stored on object storage (e.g., S3).
+
+
+#### Purpose
+
+- Proxy access to the Kopia repository.
+- Manage user authentication and authorization.
+- Handle repository unlocking using repository credentials.
+- Provide TLS-secured communication.
+
+#### Spec Fields
+
+| Field                        | Type        | Description                                                                                         |
+|------------------------------|-------------|-------------------------------------------------------------------------------------------------|
+| `backupRepositoryRef`             | string (optional) | Reference to the Velero **BackupRepository (BR)** resource. Cannot be used together with `backupStorageLocationRepositoryRef`.   |
+| `backupStorageLocationRepositoryRef` | string (optional) | Reference to the **BackupStorageLocationRepository (BSLR)** resource. Cannot be used together with `backupRepositoryRef`.|
+| `userSecretRef.name`              | string (optional) | Name of a Secret containing Kopia proxy server user credentials (username/password).             |
+| `loadAffinityConfig`              | array of `LoadAffinity` (optional) | Specifies node affinity rules for the Kopia server pods.   |
+| `podResources`                   | `PodResources` (optional) | Specifies CPU/memory resource requests and limits for pods launched by the BSLS. |
+---
+
+#### Status Fields
+
+| Field              | Type    | Description                                  |
+|--------------------|---------|----------------------------------------------|
+| `phase`            | string  | Current lifecycle phase, e.g., `Ready`.      |
+| `endpoint`         | string  | Network address to reach the Kopia Server.  |
+| `tlsFingerprint`   | string  | SHA-256 fingerprint of the server TLS cert. |
+| `conditions`       | array   | List of current status conditions/events.   |
+
+---
+
+```yaml
+# Example BSLS CRD specification
+apiVersion: oadp.openshift.io/v1alpha1
+kind: BackupStorageLocationServer
```

Comment:
What container image will be used for this backup server ? Official kopia image or some custom OADP extended image ? 
I think we also add an image override via DPA for this server.